### PR TITLE
Enable postgres_exporter by default

### PR DESCRIPTION
**Pull request by nhudson**
Enable postgres_exporter by default

## Description

We want to enable the postgres_exporter provided from timescaledb-single Helm chart by default.  This will allow us to collect metrics from the DB by default.
- Add in postgres-exporter mixin alerts and Grafana dashboard
- Fix and merge how we build and generate mixin dashboards and alerts

## Type of change

*What type of changes does your code introduce to tobs? Put an `x` in the box that apply.*

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

### CODE REVIEW

**Comment by nhudson** [comment]
The mixin Dashboard seems to have some issues specifically around thhe datasource, as it's not really labeled in a consistent way. See https://github.com/prometheus-community/postgres_exporter/pull/647 for a bit more context.

File: chart/templates/postgres-exporter-promrule.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# DO NOT EDIT THIS FILE.
+# This file is generated from scripts/sync-postgres-exporter-promrule.sh
```
Comment (paulfantom) [inline chart/templates/postgres-exporter-promrule.yaml:2]: Is this correct? Isn't `alerts/postgres-exporter-alerts.yaml` the generated one?

File: chart/values.yaml [hunk 2]
```diff
@@ -188,13 +200,13 @@ kube-prometheus-stack:
...
         # - {protocol}://{host}:{port}/{endpoint}
-        - url: "http://{{ .Release.Name }}-promscale-connector.{{ .Release.Namespace }}.svc:9201/read"
+        - url: "http://{{ .Release.Name }}-promscale.{{ .Release.Namespace }}.svc:9201/read"
           readRecent: true
 
       # The remote_write spec configuration for Prometheus.
       # ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#remotewritespec
       remoteWrite:
-        - url: "http://{{ .Release.Name }}-promscale-connector.{{ .Release.Namespace }}.svc:9201/write"
+        - url: "http://{{ .Release.Name }}-promscale.{{ .Release.Namespace }}.svc:9201/write"
```
Comment (paulfantom) [inline chart/values.yaml:209]: We're fixing those in https://github.com/timescale/tobs/pull/521

File: chart/values.yaml [hunk 3]
```diff
@@ -299,7 +312,7 @@ kube-prometheus-stack:
         port: 5432
     jaeger:
       # Endpoint for integrating jaeger datasource in grafana. This should point to HTTP endpoint, not gRPC.
-      promscaleTracesQueryEndPoint: "{{ .Release.Name }}-promscale-connector.{{ .Release.Namespace }}.svc:9201"
+      promscaleTracesQueryEndPoint: "{{ .Release.Name }}-promscale.{{ .Release.Namespace }}.svc:9201"
```
Comment (paulfantom) [inline chart/values.yaml:315]: We're fixing those in https://github.com/timescale/tobs/pull/521

File: chart/values.yaml [hunk 4]
```diff
@@ -188,13 +200,13 @@ kube-prometheus-stack:
       # ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#remotereadspec
       remoteRead:
         # - {protocol}://{host}:{port}/{endpoint}
-        - url: "http://{{ .Release.Name }}-promscale-connector.{{ .Release.Namespace }}.svc:9201/read"
+        - url: "http://{{ .Release.Name }}-promscale.{{ .Release.Namespace }}.svc:9201/read"
```
Comment (paulfantom) [inline chart/values.yaml:203]: We're fixing those in https://github.com/timescale/tobs/pull/521

File: chart/values.yaml [hunk 5]
```diff
@@ -284,7 +297,7 @@ kube-prometheus-stack:
         # By default url of data source is set to ts-prom connector instance
         # deployed with this chart. If a connector isn't used this should be
         # set to the prometheus-server.
-        url: "http://{{ .Release.Name }}-promscale-connector.{{ .Release.Namespace }}.svc:9201"
+        url: "http://{{ .Release.Name }}-promscale.{{ .Release.Namespace }}.svc:9201"
```
Comment (paulfantom) [inline chart/values.yaml:300]: We're fixing those in https://github.com/timescale/tobs/pull/521

File: scripts/sync-postgres-exporter-mixin.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+fi
+mkdir -p tmp
+git clone --depth 1 https://github.com/prometheus-community/postgres_exporter.git tmp/postgres_exporter
+
+# This mixin requires a few tools to be installed, we need to check for those
+# and print out that they need to be installed if they are not.
+
+if [ ! -x "$(command -v go)" ]; then
+	echo "go is not installed, please install it"
+	exit 0
```
Comment (paulfantom) [inline scripts/sync-postgres-exporter-mixin.sh:21]:
```suggestion
	exit 1
```
we shouldn't exit with success status code when the script cannot be executed properly

File: scripts/sync-postgres-exporter-mixin.sh [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+if [ ! -x "$(command -v mixtool)" ]; then
+	echo "mixtool is not installed, please install it"
+	echo "https://github.com/prometheus-community/postgres_exporter/tree/master/postgres_mixin"
+	exit 0
```
Comment (paulfantom) [inline scripts/sync-postgres-exporter-mixin.sh:27]:
```suggestion
	exit 1
```
same as above

File: scripts/sync-postgres-exporter-mixin.sh [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+# This seems like the most straightforward way to replace the datasource in the
+# generated dashboard for now. We can revisit this if we find a better way to
+# replace the generated datasource.
+for file in `ls ${mixin}/dashboards_out/*.json`
+do
+	cp -r $file chart/dashboards/$(basename $file)
+	sed -i.orig 's/\"datasource\": .*$/\"datasource\": {\n    \"type\": \"prometheus\",\n    \"uid\": \"dc08d25c8f267b054f12002f334e6d3d32a853e4\"\n }, /g' chart/dashboards/$(basename $file)
+	find chart/dashboards/ -name '*.orig' -exec rm -f {} \;
+done
```
Comment (paulfantom) [inline scripts/sync-postgres-exporter-mixin.sh:49]: We do have this logic in https://github.com/timescale/tobs/blob/main/scripts/sync-dashboards.sh, maybe we can extract it to some `make` target or some other script and align both. WDYT?

File: chart/templates/postgres-exporter-promrule.yaml (hunk 1 above)
Comment (nhudson) [reply chart/templates/postgres-exporter-promrule.yaml:2]: Yeah it's left over from some older code, I will remove it.

File: scripts/sync-postgres-exporter-mixin.sh (hunk 8 above)
Comment (nhudson) [reply scripts/sync-postgres-exporter-mixin.sh:49]: Yeah syncing both scripts into one and adding a make target sounds like a good idea.  I will get that knocked out.

File: .github/workflows/dashboards.yml [hunk 9]
```diff
@@ -17,6 +20,14 @@ jobs:
...
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.golang-version }}
+
     - name: Run Dashboard synchronizer
       run: |
-        ./scripts/sync-dashboards.sh && git diff --exit-code
+        go install github.com/monitoring-mixins/mixtool/cmd/mixtool@latest && \
+        go install github.com/google/go-jsonnet/cmd/jsonnetfmt@latest && \
+        ./scripts/sync-mixins.sh && \
```
Comment (nhudson) [inline .github/workflows/dashboards.yml:32]: Use `make` target instead of calling the script directly.

**Review by paulfantom — APPROVED** [review]

**Pull request merged**